tests/unit_tests/test_subtensor.py

```diff
@@ -475,3 +474,1710 @@ def test_weights_rate_limit_success_calls(subtensor, mocker):
+    spy_balance_from_rao = mocker.spy(Balance, "from_rao")
+
+    # Call
+    result = subtensor.get_total_stake_for_hotkey(ss58_address=fake_ss58_address)
+
+    # Assertions
+    subtensor.query_subtensor.assert_called_once_with(
+        "TotalHotkeyStake", None, [fake_ss58_address]
+    )
+    spy_balance_from_rao.assert_not_called()
+    # if we change the methods logic in the future we have to be make sure tha returned type is correct
+    assert isinstance(subtensor.query_subtensor.return_value, object)
+    assert not hasattr(result, "value")
+
+
+# `get_total_stake_for_coldkey` tests
+def test_get_total_stake_for_coldkey_success(subtensor, mocker):
+    """Tests successful retrieval of total stake for coldkey."""
+    # Prep
+    subtensor.query_subtensor = mocker.MagicMock(return_value=mocker.MagicMock(value=1))
+    fake_ss58_address = "12bzRJfh7arnnfPPUZHeJUaE62QLEwhK48QnH9LXeK2m1iZU"
+    spy_balance_from_rao = mocker.spy(Balance, "from_rao")
+
+    # Call
+    result = subtensor.get_total_stake_for_coldkey(ss58_address=fake_ss58_address)
+
+    # Assertions
+    subtensor.query_subtensor.assert_called_once_with(
+        "TotalColdkeyStake", None, [fake_ss58_address]
+    )
+    spy_balance_from_rao.assert_called_once()
+    # if we change the methods logic in the future we have to be make sure tha returned type is correct
+    assert isinstance(result, Balance)
+
+
+def test_get_total_stake_for_coldkey_not_result(subtensor, mocker):
+    """Tests retrieval of total stake for coldkey when no result is returned."""
+    # Prep
+    subtensor.query_subtensor = mocker.MagicMock(return_value=None)
+    fake_ss58_address = "12bzRJfh7arnnfPPUZHeJUaE62QLEwhK48QnH9LXeK2m1iZU"
+    spy_balance_from_rao = mocker.spy(Balance, "from_rao")
+
+    # Call
+    result = subtensor.get_total_stake_for_coldkey(ss58_address=fake_ss58_address)
+
+    # Assertions
+    subtensor.query_subtensor.assert_called_once_with(
+        "TotalColdkeyStake", None, [fake_ss58_address]
+    )
+    spy_balance_from_rao.assert_not_called()
+    # if we change the methods logic in the future we have to be make sure tha returned type is correct
+    assert isinstance(result, type(None))
+
+
+def test_get_total_stake_for_coldkey_not_value(subtensor, mocker):
+    """Tests retrieval of total stake for coldkey when no value attribute is present."""
+    # Prep
+    subtensor.query_subtensor = mocker.MagicMock(return_value=object)
+    fake_ss58_address = "12bzRJfh7arnnfPPUZHeJUaE62QLEwhK48QnH9LXeK2m1iZU"
+    spy_balance_from_rao = mocker.spy(Balance, "from_rao")
+
+    # Call
+    result = subtensor.get_total_stake_for_coldkey(ss58_address=fake_ss58_address)
+
+    # Assertions
+    subtensor.query_subtensor.assert_called_once_with(
+        "TotalColdkeyStake", None, [fake_ss58_address]
+    )
+    spy_balance_from_rao.assert_not_called()
+    # if we change the methods logic in the future we have to be make sure tha returned type is correct
```

Comment:
"the", but I don't feel this comment is necessary.

Reply:
done